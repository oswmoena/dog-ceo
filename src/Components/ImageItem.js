import React, { useState, useCallback } from 'react'
import { makeStyles, createStyles } from '@mui/styles';
import { ImageList, ImageListItem, ImageListItemBar } from '@mui/material';
import ImageViewer from 'react-simple-image-viewer';

const useStyles = makeStyles((theme) =>
    createStyles({
        root: {
            display: 'flex',
            flexWrap: 'wrap',
            justifyContent: 'space-around',
            overflow: 'hidden',
        },
        imageList: {
            flexWrap: 'nowrap',
            transform: 'translateZ(0)',
        },
        title: {
            color: '#FFFFFF',
        },
        titleBar: {
            background:
                'linear-gradient(to top, rgba(0,0,0,0.7) 0%, rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)',
        },
    }));

export const ImageItem = ({ images, title }) => {
    const [currentImage, setCurrentImage] = useState(0);
    const [isViewerOpen, setIsViewerOpen] = useState(false);
    const classes = useStyles();

    const openImageViewer = useCallback((index) => {
        setCurrentImage(index);
        setIsViewerOpen(true);
    }, []);

    const closeImageViewer = () => {
        setCurrentImage(0);
        setIsViewerOpen(false);
    };

    return (
        <div className={classes.root}>
            <ImageList className={classes.imageList} cols={3}>
                {images.map((item, index) => (
                    <ImageListItem key={item}>
                        <img
                            src={item}
                            onClick={() => openImageViewer(index)}
                            width="300"
                            key={index}
                            style={{ margin: '2px' }}
                            alt={title}
                        />
                        <ImageListItemBar
                            title={title}
                            classes={{
                                root: classes.titleBar,
                                title: classes.title,
                            }}
                            loading="lazy"
                        />
                    </ImageListItem>
                ))}
            </ImageList>
            {isViewerOpen && (
                <ImageViewer
                    src={images}
                    currentIndex={currentImage}
                    disableScroll={false}
                    closeOnClickOutside={true}
                    onClose={closeImageViewer}
                />
            )}
        </div>
    )




}
