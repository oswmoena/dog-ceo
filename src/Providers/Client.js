import axios from 'axios'

const client = axios.create({
    baseURL: `https://dog.ceo/api`,
})

export const response = async (options) => {
    options.headers = {
        'Content-Type': 'application/json',
    }

    return await client(options)
        .then(response => {
            if (response.data) {
                const { data } = response
                return data
            } else {
                return response
            }
        })
        .catch(error => {
            console.error('error', error)
        })
}