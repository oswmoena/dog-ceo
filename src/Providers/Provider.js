import { response } from './Client'

export const getAllBreeds = () => {
    const options = {
        url: `/breeds/list/all`,
    }
    return response(options)
}

export const getBreed = (breed) => {
    const options = {
        url: `/breed/${breed}/images/random/3`,
    }
    return response(options)
}

export const getSubBreed = (breed, subbreed) => {
    const options = {
        url: `/breed/${breed}/${subbreed}/images/random/3`,
    }
    return response(options)
}