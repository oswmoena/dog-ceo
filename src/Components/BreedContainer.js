import React, { useEffect, useState } from 'react'
import { getBreed } from '../Providers/Provider'
import { ImageItem } from './ImageItem'
import AutorenewIcon from '@mui/icons-material/Autorenew';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme) => createStyles({
    spacing: {
        display: 'flex',
        justifyContent: 'center',
        padding: '5%',
        fontSize: '20px'
    },
    'animatedItem': {
        animation: `$spin 2s infinite linear`,
    },
    '@global': {
        '@keyframes spin': {
            '0%': {
                transform: 'rotate(0deg)',
            },
            '100%': {
                transform: 'rotate(360deg)',
            },
        },
    },
}))

export const BreedContainer = ({ breed }) => {
    const classes = useStyles();
    const [error, setError] = useState(false)
    const [breedImage, setBreedImage] = useState([])

    useEffect(() => {
        getBreedSelected()
    }, []) // eslint-disable-line react-hooks/exhaustive-deps

    const getBreedSelected = async () => {
        const response = await getBreed(breed.toLowerCase());
        if (response) {
            setError(false)
            setBreedImage(response.message)
        } else {
            setError(true)
        }
    }

    if (error) {
        return (
            <div className={classes.spacing}>
                <p>Error en la comunicaci√≥n con el servidor</p>
            </div>
        )
    }

    return breedImage.length > 0
        ? (
            <ImageItem images={breedImage} title={breed} />
        ) : (
            <div className={classes.spacing}>
                <AutorenewIcon className={classes.animatedItem} />
            </div>
        )


}
