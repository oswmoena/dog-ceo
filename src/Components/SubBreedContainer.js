import React, { useState, useEffect } from 'react'
import { getSubBreed } from '../Providers/Provider'
import { ImageItem } from './ImageItem'

export const SubBreedContainer = ({ breed, subbreed }) => {
    const [subBreedImage, setSubBreedImage] = useState([])

    useEffect(() => {
        getSubBreedSelected()
    }, [])

    const getSubBreedSelected = async () => {
        const response = await getSubBreed(breed.toLowerCase(), subbreed.toLowerCase());
        setSubBreedImage(response.message)
    }

    return subBreedImage.length > 0 ? (
        <ImageItem images={subBreedImage} title={subbreed} />
    ) : (
        <h3>Cargando...</h3>
    )
}
