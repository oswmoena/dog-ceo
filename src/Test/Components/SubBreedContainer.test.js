import React from 'react';
import { shallow } from 'enzyme'
import { SubBreedContainer } from '../../Components/SubBreedContainer';
import { getBreed } from '../../Providers/Provider';


describe('Test on SubBreed Container component', () => {
    test('Show SubBreedContainer correctly', () => {
        const wrapper = shallow(<SubBreedContainer />);
        expect(wrapper).toMatchSnapshot()
    })

    test('get sub breed selected', async () => {
        const breed = await getBreed('hound', 'english')

        const { status, message } = breed

        expect(status).toBe('success')
        expect(message.length).toBe(3)

    })

    test('get only 3 sub breeds', async () => {
        const breed = await getBreed('hound', 'english')

        const { status, message } = breed

        expect(message.length).toBe(3)

    })
})