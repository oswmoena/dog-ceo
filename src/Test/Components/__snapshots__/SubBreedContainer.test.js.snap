// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on SubBreed Container component Show SubBreedContainer correctly 1`] = `
<div
  className="makeStyles-spacing-1"
>
  <Memo(ForwardRef(AutorenewIcon)) />
</div>
`;
